cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/SubqueryExpressionsTest.scala

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright (c) "Neo4j"
+ * Neo4j Sweden AB [http://neo4j.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.cypher.docgen
+
+import org.neo4j.cypher.docgen.tooling.{DocBuilder, DocumentingTest, ResultAssertions}
+
+class SubqueryExpressionsTest extends DocumentingTest {
+  override def outputPath = "target/docs/dev/ql/"
+
+  override def doc = new DocBuilder {
+    doc("Subquery expressions", "cypher-subquery-expressions")
+    initQueries(
+      """CREATE
+        #(andy:Swedish:Person {name: 'Andy', age: 36, belt: 'white'}),
+        #(timothy:Person {name: 'Timothy', age: 25, address: 'Sweden/Malmo'}),
+        #(peter:Person {name: 'Peter', age: 35, email: 'peter_n@example.com'}),
+        #(andy)-[:KNOWS {since: 2012}]->(timothy),
+        #(andy)-[:KNOWS {since: 1999}]->(peter),
+        #(andy)-[:HAS_DOG {since: 2016}]->(:Dog {name:'Andy'}),
+        #(fido:Dog {name:'Fido'})<-[:HAS_DOG {since: 2010}]-(peter)-[:HAS_DOG {since: 2018}]->(:Dog {name:'Ozzy'}),
+        #(fido)-[:HAS_TOY]->(:Toy{name:'Banana'})"""".stripMargin('#'))
+    synopsis("Cypher has expressions that evaluate a subquery and aggregate the result in different fashions.")
+    p(
+      """* <<existential-subqueries,EXISTS subqueries>>
+        # ** <<existential-subquery-simple-case, Simple EXISTS subquery>>
+        # ** <<existential-subquery-with-where, EXISTS subquery with `WHERE` clause>>
+        # ** <<existential-subquery-nesting, Nesting EXISTS subqueries>>""".stripMargin('#'))
+    p("""Subquery expressions can appear anywhere that an expression is valid.""".stripMargin('#'))
+    p("The following graph is used for the examples below:")
+    graphViz()
+
+    section("EXISTS subqueries", "existential-subqueries") {
+      p(
+        """An EXISTS subquery can be used to find out if a specified pattern exists at least once in the data.
+          #It can be used in the same way as a path pattern but it allows you to use `MATCH` and `WHERE` clauses internally.
```

Comment:
As this was moved from the `WHERE`-clause page, it might be worth having a cross link as to how path patterns are used. Also, an `EXISTS` subquery can be used in more cases than a path pattern, so maybe we want to mention that here as well.

Reply:
I rephrased this a bit. But since there is no section about path patterns, I cannot really link to anything except this overview https://neo4j.com/docs/cypher-manual/current/syntax/expressions/ and that might not be so great.

Reply:
I found the section on path patterns again:
https://development.neo4j.dev/docs/cypher-manual/5.0/clauses/where/#filter-on-patterns